ops/model.py

```diff
@@ -1244,15 +1250,18 @@ class Relation:
             For subordinate relations, this set will include only one unit: the principal unit.
         data: A :class:`RelationData` holding the data buckets for each entity
             of a relation. Accessed via eg Relation.data[unit]['foo']
+        active: Indicates whether this relation is active (True) or inactive (False, because
+            this relation is currently part of a relation-broken event)
     """
 
     def __init__(
             self, relation_name: str, relation_id: int, is_peer: bool, our_unit: Unit,
-            backend: '_ModelBackend', cache: '_ModelCache'):
+            backend: '_ModelBackend', cache: '_ModelCache', active: bool = True):
         self.name = relation_name
         self.id = relation_id
         self.app: Optional[Application] = None
         self.units: Set[Unit] = set()
+        self.active = active
```

Comment:
Could/should we evaluate it here instead of propagating?

Reply:
Sorry I'm not sure what you mean - can you restate this?

Reply:
on the one hand it would greatly reduce the amount of code changes, on the other it is nice that code this 'deep' in ops is profoundly unaware of juju-specific envvars. It'd be nice if the only pieces of code doing envvar unpacking were in ops.main. 

How we propagate that info tastefully and in a future-proof manner, that is the question

Reply:
Yeah, it would be "simpler", but I definitely prefer doing as much env var handling as possible in `main.py`, so it's contained there. It's a bit awkward to pass it down, but I think important for testability and explicitness.